### DISCUSSION TITLE
Can't connect to remote server which is configured by laravel forge

### DISCUSSION THREAD

**Discussion by zgabievi** [spatie/ray#418] [Q&A]
**Describe the bug**
Can't connect to remote server over ssh, which is using laravel forge. I've private key connection to the server. I can connect to server using terminal but not with Ray app.

**Versions**
Ray version: 1.14.7

You can use `composer show` to get the version numbers of:
- spatie/ray package version: 1.21.2
- spatie/laravel-ray package version: 1.17.1

PHP version: 7.4.16
Laravel version: 8.35.1

**To Reproduce**
1. Go to 'Servers'
2. Add server using ip address, ssh key path and forge user
3. Try to connect

**Expected behavior**
It should connect to server.

**Screenshots**
![image](https://user-images.githubusercontent.com/1515299/113439391-873e1800-93fb-11eb-84cf-1f63736f292e.png)

**Desktop (please complete the following information):**
 - OS: macOS
 - Version: 11.2.3

**Comment by jkbennemann** [comment]
I do have almost the same issue, but not specifically with forge.

**Describe the bug**
For some reason I am not able to connect to a remote server.
On connection ray responds with information: "Ray was unable to bind to the remote server's port. Is the port in use or is someone else using ray on this server?"

The direct connection through iTerm does work for both servers though.

**Versions**
Ray version: 1.14.7

Composer: not installed

PHP version: 7.4.16

**To Reproduce**
Steps to reproduce the behavior:
reproducable with my private ssh key only.
See additional context below.

**Expected behavior**
It would expect that ray is able to connect to the server ssh.dsdae.com via port 22 and my ssh key as they are same for both servers.

**Screenshots**
First server works seamlessly:
<img width="715" alt="Bildschirmfoto 2021-04-13 um 10 14 57" src="https://user-images.githubusercontent.com/25330395/114522328-9450d780-9c43-11eb-8b66-cea431af0742.png">

Error on connecting to ssh.dsdae.com
<img width="744" alt="Bildschirmfoto 2021-04-13 um 10 06 01" src="https://user-images.githubusercontent.com/25330395/114522367-9fa40300-9c43-11eb-9058-7bf493fad80c.png">


**Desktop (please complete the following information):**
 - OS: macOS
 - Version: 11.2.3 (20D91) - BigSur
 
 **Additional context**

The "working" server does have simple iptables firewall settings in place which only allow connection through port 22, 80, 443, established once and local interface. Everything else is dropped.

The "non-working" server does have almost the same iptables firewall settings in place which only allow connection through port 22, 80, 443, established once and local interface and some very specific ones for smtp config. I even switched off the entire firewall temporarily but this does not work neither.

Because of this I don't think the firewall is 

```bash
Working connection through iTerm via ssh for server which ray is able to connect to.
{10:03}~ ➭ ssh jakob@lbXXXXXXXXXXXXXXn.de ##working
{10:03}~ ➭ ssh -i ~/.ssh/id_rsa -p 22 jakob@lbXXXXXXXXXXXXXXn.de ##working
Welcome to Ubuntu
Last login: Tue Apr 13 10:03:59 2021 from XXXXXXXXXXXXXXX
jakob@lb-01:~$


##Working connection through iTerm via ssh for server which ray is not able to connect to.
{10:01}~ ➭ ssh wp@ssh.dsdae.com  ##working
{10:01}~ ➭ ssh -i ~/.ssh/id_rsa -p 22 wp@ssh.dsdae.com ##working
Last login: Tue Apr 13 10:01:42 2021 from XXXXXXXXXXXXXXX
[21-04-13 10:06]wp@ssh.dsdae.com:~%

**Comment by chimperial** [comment]
Maybe I'm facing the same issue.

<img width="269" alt="image" src="https://user-images.githubusercontent.com/86779613/165214024-0fb649d4-0630-4569-bf5f-c2a8d67cde5d.png">

It happens when my wifi disconnected due to some reasons. Is there any way to kill the old ray port on the remote server?

**Reply by jmslbam** [reply]
I had the same thing @ 0xSonaMaka said, not what the original discussion was about... I switched to a different Wi-Fi in the mids of debugging with Ray on a server.  So it all worked and then after switching it stopped working. 

I had to wait around 15 tot 20 minutes before I could reconnect again, so just grab a coffee :)

**Comment by cpl-repo** [comment]
Hey Spatie can you fix this? No I don't go for coffee after switching wifi, this just has to work ;)
My internet connection is sometimes unstable and therefore in the middle of working and debugging Laravel ray stops to work with this error.

**Comment by jmslbam** [comment]
@cpl-repo Quite a rude way of asking for a favor for somebody else to fix your laggy internet problem. Although I do know you and I pay for a license (I rather see it as a donate to support the future development of this so super usefull product) , which saves you ton's of time and there for money.

But that's my personal Open Source experience throughout the years.
